### PR TITLE
feat(proposer): improve sync progress check

### PR DESCRIPTION
**Pull request by davidtaikocha**

### CODE REVIEW

File: pkg/rpc/methods.go [hunk 1]
```diff
@@ -224,6 +225,53 @@ func (c *Client) L2AccountNonce(
 	return uint64(result), err
 }
 
+// L2SyncProgress represents the sync progress of a L2 execution engine, `ethereum.SyncProgress` is used to check
+// the sync porgress of verified blocks, and block IDs is used to check sync progress of pending blocks.
```
Comment (alexshliu) [inline pkg/rpc/methods.go:229]:
```suggestion
// the sync progress of verified blocks, and block IDs are used to check the sync progress of pending blocks.
```

File: pkg/rpc/methods.go [hunk 2]
```diff
@@ -224,6 +225,53 @@ func (c *Client) L2AccountNonce(
...
+		headL1Origin, err := c.L2.HeadL1Origin(ctx)
+		if err != nil {
+			return err
+		}
+		currentBlockID = headL1Origin.BlockID
+		return nil
+	})
+
+	return &L2SyncProgress{SyncProgress: progress, CurrentBlockID: currentBlockID, HighestBlockID: highestBlockID},
+		g.Wait()
```
Comment (alexshliu) [inline pkg/rpc/methods.go:272]: Maybe the following code is more readable:

```go 
func (c *Client) L2ExecutionEngineSyncProgress(ctx context.Context) (*L2SyncProgress, error) {
	var err error
	p := new(L2SyncProgress)

	g, ctx := errgroup.WithContext(ctx)

	g.Go(func() error {
		p.SyncProgress, err = c.L2.SyncProgress(ctx)
		return err
	})

	g.Go(func() error {
		stateVars, err := c.GetProtocolStateVariables(nil)
		if err != nil {
			return err
		}
		p.HighestBlockID = new(big.Int).SetUint64(stateVars.NextBlockID - 1)
		log.Info("state vars", "vars", stateVars, "highest", p.HighestBlockID)
		return nil
	})

	g.Go(func() error {
		headL1Origin, err := c.L2.HeadL1Origin(ctx)
		if err != nil {
			return err
		}
		p.CurrentBlockID = headL1Origin.BlockID
		return nil
	})

	if err := g.Wait(); err != nil {
		return nil, err
	}
	return p, nil
}
```

File: pkg/rpc/methods.go (hunk 1 above)
Comment (davidtaikocha) [reply pkg/rpc/methods.go]: changed

File: pkg/rpc/methods.go (hunk 2 above)
Comment (davidtaikocha) [reply pkg/rpc/methods.go]: changed

**Review by alexshliu — APPROVED** [review]

**Pull request merged**